frontend/spec/controllers/spree/checkout_controller_spec.rb

```diff
@@ -429,4 +429,46 @@ def post_address
       post :update, params: { state: "payment" }
     }.to change { order.line_items.to_a.size }.from(1).to(0)
   end
+
+  context 'trying to apply a coupon code' do
+    let(:order) do
+      create(:order_with_line_items).tap do |order|
+        until order.state == 'payment'
```

Comment:
Should this be a factory trait of some sort?

Reply:
yes, I can just do `create(:order_with_line_items, state: 'payment')`